spec/models/story_parser_spec.rb

```diff
@@ -193,21 +193,67 @@ class StoryParser
     end
 
     it "raises an exception when the external author name is not provided" do
-      expect {
+      expect do
         @sp.parse_author("", nil, "author@example.com")
-      }.to raise_exception(StoryParser::Error) { |e| expect(e.message).to eq("No author name specified") }
+      end.to raise_exception(StoryParser::Error, "No author name specified")
     end
 
     it "raises an exception when the external author email is not provided" do
-      expect {
+      expect do
         @sp.parse_author("", "Author Name", nil)
-      }.to raise_exception(StoryParser::Error) { |e| expect(e.message).to eq("No author email specified") }
+      end.to raise_exception(StoryParser::Error, "No author email specified")
     end
 
     it "raises an exception when neither the external author name nor email is provided" do
-      expect {
+      expect do
         @sp.parse_author("", nil, nil)
-      }.to raise_exception(StoryParser::Error) { |e| expect(e.message).to eq("No author name specified\nNo author email specified") }
+      end.to raise_exception(StoryParser::Error, "No author name specified\nNo author email specified")
+    end
+
+    it "gives the same external author object for the same email" do
+      res1 = @sp.parse_author("", "Author Name", "author@example.com")
```

Comment:
I think it's possible to replace the instance variable with ``let``. Not sure if it's a good idea to do it now, because that would be an unrelated refactoring that would touch most of the methods here.